packages/gatsby-plugin-gatsby-cloud/src/build-headers-program.js

```diff
@@ -273,9 +273,19 @@ const applyCachingHeaders = (
     return headers
   }
 
-  const chunks = Array.from(pluginData.pages.values()).map(
-    page => page.componentChunkName
-  )
+  let chunks = []
+  // Gatsby v3.5 added componentChunkName to store().components
+  // So we prefer to pull chunk names off that as it gets very expensive to loop
+  // over large numbers of pages.
+  const isComponentChunkSet = !!pluginData.components.entries().next().value[1]
```

Comment:
are you sure this can't be nullable?

Reply:
Like the component object itself? I'm not sure what you mean. Do you see a possible problem?

Reply:
I guess it's possible to have no pages in which case yeah there'd be no entries. I'll add some `?` just in case.

Reply:
https://github.com/gatsbyjs/gatsby/pull/31547/commits/dc6f154ed2758c335b3a4e1c424bde53736b5bb5